src/main/java/me/ryanhamshire/GriefPrevention/PlayerEventHandler.java

```diff
@@ -1335,6 +1339,47 @@ public void onPlayerInteractEntity(PlayerInteractEntityEvent event)
         }
     }
 
+    //when a thrown projectile hit its target
+    @EventHandler(priority = EventPriority.LOWEST)
+    public void onProjectileHit(ProjectileHitEvent event)
+    {
+        //if the projectile is an egg and the thrower is a player
+        if (event.getEntity() instanceof Egg && event.getEntity().getShooter() instanceof Player)
+        {
+            Player player = (Player) event.getEntity().getShooter();
+            PlayerData playerData = this.dataStore.getPlayerData(player.getUniqueId());
+            Claim claim = null;
+
+            //allow egg hit if player is in ignore claims mode
+            if (playerData.ignoreClaims) return;
+
+            //if the egg hit a block
+            if (event.getHitBlock() != null)
+            {
+                claim = this.dataStore.getClaimAt(event.getHitBlock().getLocation(), false, playerData.lastClaim);
+            }
+            //if the egg hit an entity
+            else if (event.getHitEntity() != null)
+            {
+                claim = this.dataStore.getClaimAt(event.getHitEntity().getLocation(), false, playerData.lastClaim);
+            }
+
+            if (claim != null && claim.allowContainers(player) != null)
+            {
+                String message = this.instance.dataStore.getMessage(Messages.NoContainersPermission, claim.getOwnerName());
+
+                if (player.hasPermission("griefprevention.ignoreclaims"))
+                {
+                    message += "  " + instance.dataStore.getMessage(Messages.IgnoreClaimsAdvertisement);
+                }
+
+                GriefPrevention.sendMessage(player, TextMode.Err, message);
+                player.getInventory().addItem(((Egg) event.getEntity()).getItem());
+                return;
+            }
+        }
+    }
```

Comment:
Don't see anywhere where you stop the egg from hatching. Am assuming you haven't had an opportunity to test yet.

Reply:
Unfortunately this event can't be cancelled. Though I forgot to remove the egg entity, that is now fixed. And yes, I haven't tested it